src/src_main/crypto_tc.c

```diff
@@ -257,14 +272,41 @@ int32_t Crypto_TC_ApplySecurity(const uint8_t* p_in_frame, const uint16_t in_fra
             // TODO: If ARSN is transmitted in the SHSNF field (as in CMAC... don't double count those bytes)
             *p_enc_frame_len = temp_tc_header.fl + 1 + 2 + sa_ptr->shivf_len + sa_ptr->shsnf_len + sa_ptr->shplf_len +
                                TC_PAD_SIZE + sa_ptr->stmacf_len;
-            new_enc_frame_header_field_length = (*p_enc_frame_len) - 1;
+            new_enc_frame_header_field_length = (*p_enc_frame_len) - 1;            
             break;
         case SA_ENCRYPTION:
             // Ingest length + spi_index (2) + shivf_len (varies) + shsnf_len (varies)
             //   + shplf_len + arsn_len + pad_size
             *p_enc_frame_len = temp_tc_header.fl + 1 + 2 + sa_ptr->shivf_len + sa_ptr->shsnf_len + sa_ptr->shplf_len +
-                               sa_ptr->arsn_len + TC_PAD_SIZE;
+                               sa_ptr->arsn_len + TC_PAD_SIZE; //should point to shplf_len
```

Comment:
How come this is TC_PAD_SIZE here? Doesn't this need to be the value of the "padding_needed" below?

Reply:
Oh OK, you add it below -- and this works because TC_PAD_SIZE is 0 ... Maybe we should just remove TC_PAD_SIZE entirely now that we're actually handling it.

Reply:
Yeah, I think it can probably go away.

Reply:
Removed

Reply:
Confirmed removed in commit 7b13732